src/backend/distributed/planner/recursive_planning.c

```diff
@@ -314,6 +315,25 @@ RecursivelyPlanSubqueriesAndCTEs(Query *query, RecursivePlanningContext *context
 		RecursivelyPlanAllSubqueries((Node *) query->jointree->quals, context);
 	}
 
+	if (query->havingQual != NULL)
+	{
+		List *havingSublinks = NIL;
+		ExtractSublinkWalker(query->havingQual, &havingSublinks);
+
+		SubLink *havingSublink;
+		foreach_ptr(havingSublink, havingSublinks)
+		{
+			if (ContainsReferencesToOuterQuery(castNode(Query, havingSublink->subselect)))
+			{
+				return DeferredError(ERRCODE_FEATURE_NOT_SUPPORTED,
+									 "Cannot recursively plan HAVING with references to outer query",
```

Comment:
This probably wouldn't be clear to users unfamiliar with recursive planning. Maybe "Subqueries in HAVING cannot refer to the outer query".